rslib/src/scheduler/answering/mod.rs

```diff
@@ -670,4 +702,85 @@ mod test {
 
         Ok(())
     }
+
+    #[test]
+    fn elapsed_time() -> Result<()> {
+        let mut col = Collection::new();
+        let mut conf = col.get_deck_config(DeckConfigId(1), false)?.unwrap();
+        let nt = col.get_notetype_by_name("Basic")?.unwrap();
+        let mut note = nt.new_note();
+        // Need to set col age for interday learning test, arbitrary
+        col.storage
+            .db
+            .execute_batch("update col set crt=1686045847")?;
+        // Fails when near cutoff since it assumes inter- and intraday learning
+        if col.timing_today()?.near_cutoff() {
+            return Ok(());
+        }
+        col.add_note(&mut note, DeckId(1))?;
+        // 5942.7 minutes for just over four days
+        conf.inner.learn_steps = vec![1.0, 10.5, 15.0, 20.0, 5942.7];
+        col.storage.update_deck_conf(&conf)?;
+
+        // Intraday learning, review same day
+        let expected_elapsed_time = 662;
+        let post_answer = col.answer_good();
+        let card = col.storage.get_card(post_answer.card_id)?.unwrap();
+        let shift_due_time = card.due - expected_elapsed_time;
+        elapsed_time_helper(&mut col, shift_due_time, post_answer, expected_elapsed_time)?;
+
+        // Intraday learning, learn ahead
+        let expected_elapsed_time = 212;
+        let post_answer = col.answer_good();
+        let card = col.storage.get_card(post_answer.card_id)?.unwrap();
+        let shift_due_time = card.due - expected_elapsed_time;
+        elapsed_time_helper(&mut col, shift_due_time, post_answer, expected_elapsed_time)?;
+
+        // Intraday learning, review two (and some) days later
+        let expected_elapsed_time = 184092;
+        let post_answer = col.answer_good();
+        let card = col.storage.get_card(post_answer.card_id)?.unwrap();
+        let shift_due_time = card.due - expected_elapsed_time;
+        elapsed_time_helper(&mut col, shift_due_time, post_answer, expected_elapsed_time)?;
+
+        // Interday learning four (and some) days, review three days late
+        let expected_elapsed_time = -7;
+        let post_answer = col.answer_good();
+        let now = TimestampSecs::now();
+        let timing = col.timing_for_timestamp(now)?;
+        let col_age = timing.days_elapsed as i32;
+        let shift_due_time = col_age - 3; // Three days late
+        elapsed_time_helper(&mut col, shift_due_time, post_answer, expected_elapsed_time)?;
+
+        Ok(())
+    }
+
+    fn elapsed_time_helper(
```

Comment:
A name that better describes what it does (e.g. assert_elapsed_times_are_approx_equal) would be clearer.